templates/react/packages/client/src/mud/setupNetwork.ts

```diff
@@ -38,6 +54,12 @@ export async function setupNetwork() {
 
   const publicClient = createPublicClient(clientOptions);
 
+  /*
+   * Create an observable for contract writes that we can
+   * pass into MUD dev tools for transaction observability.
+   */
+  const write$ = new Subject<ContractWrite>();
```

Comment:
just a bit of context on this approach:

my ideal would be that dev tools would export something that we could "write to" instead of passing in a write stream to dev tools, but when I initially tried this, it got a bit hairy with bundling/code splitting, where an instance of a script running on the page may not have context on the variables created by another part of the code using the same-ish variables

would be curious if you have other ideas for approaching this

Reply:
I don't have a good idea for now. 🤔 

Is the `transportObserver` approach closer to your ideal?

Reply:
I think it might be tricky to use the transport observer approach because we'd have to somehow tie the simulation request (`eth_call`) to the transaction (`eth_sendTransaction` and `eth_sendRawTransaction`).